native/src/main/java/io/ballerina/stdlib/grpc/Message.java

```diff
@@ -1459,42 +1477,18 @@ private String anyMessageTypeNameFromTypeUrl(String typeUrl) {
         return types[types.length - 1].trim();
     }
 
-    private Descriptors.Descriptor findFieldDescriptorFromTypeUrl(String typeName) {
-
-        if (typeName.startsWith("google.protobuf")) {
-            return findGoogleDescriptorFromName(typeName);
-        } else {
-            return MessageRegistry.getInstance().getFileDescriptor().findMessageTypeByName(typeName);
-        }
-    }
-
     private Descriptors.Descriptor findFieldDescriptorFromTypeUrl(String messageName, Object value) {
 
         if (messageName.startsWith("google.protobuf")) {
             return findGoogleDescriptorFromName(messageName);
         } else if (value instanceof BMap) {
             BMap<BString, Object> bMap = (BMap<BString, Object>) value;
             RecordType recordType = (RecordType) bMap.getType();
-            if (isDescriptorAnnotationAvailable(recordType)) {
+            if (isDescriptorAnnotationAvailable(recordType) || isDescriptorAnnotationAvailable((RecordType)
+                    ValueCreator.createRecordValue(recordType.getPackage(), messageName).getType())) {
                 try {
                     return getDescriptorFromRecord(recordType, messageName);
-                } catch (InvalidProtocolBufferException e) {
-                    return MessageRegistry.getInstance().getFileDescriptor().findMessageTypeByName(messageName);
-                }
-            }
-        }
-        return MessageRegistry.getInstance().getFileDescriptor().findMessageTypeByName(messageName);
-    }
-
-    private Descriptors.Descriptor findFieldDescriptorFromTypeUrl(String messageName, RecordType recordType) {
-
-        if (messageName.startsWith("google.protobuf")) {
-            return findGoogleDescriptorFromName(messageName);
-        } else if (isDescriptorAnnotationAvailable(recordType)) {
-            try {
-                return getDescriptorFromRecord(recordType, messageName);
-            } catch (InvalidProtocolBufferException e) {
-                return MessageRegistry.getInstance().getFileDescriptor().findMessageTypeByName(messageName);
+                } catch (InvalidProtocolBufferException e) { }
```

Comment:
Please add a comment here

Reply:
Added in https://github.com/ballerina-platform/module-ballerina-grpc/pull/961/commits/106ae179a53549ba2f9358169b4692daaf4a0c57